src/kernel/include/shared/cos_types.h

```diff
@@ -223,40 +226,55 @@ captbl_idsize(cap_t c)
  * LLBooter initial captbl setup:
  * 0 = sret,
  * 1-3 = nil,
- * 4-5 = this captbl,
- * 6-7 = our pgtbl root,
- * 8-11 = our component,
- * 12-13 = vm pte for booter
- * 14-15 = untyped memory pgtbl root,
- * 16-17 = vm pte for physical memory,
- * 18-19 = km pte,
- * 20-21 = comp0 captbl,
- * 22-23 = comp0 pgtbl root,
- * 24-27 = comp0 component,
- * 28~(20+2*NCPU) = per core alpha thd
+ * 4-7 = memory access fault handler,
+ * 8-11 = divide by zero fault handler,
+ * 12-15 = breakpoint fault handler,
+ * 16-19 = invalid instruction fault handler,
+ * 20-23 = invstk overflow and underflow fault handler,
+ * 24-27 = component does not exist fault handler,
+ * 28-31 = fault handler does not exist fault handler,
+ * 32-33 = this captbl,
+ * 34-35 = our pgtbl root,
+ * 36-39 = our component,
+ * 40-41 = vm pte for booter
+ * 42-43 = untyped memory pgtbl root,
+ * 44-45 = vm pte for physical memory,
+ * 46-47 = km pte,
+ * 48-49 = comp0 captbl,
+ * 50-51 = comp0 pgtbl root,
+ * 52-55 = comp0 component,
```

Comment:
Just another nitpick: There are some of these caps that can be removed from the table.. And some that only need to be part of the booter.. If we can somehow rearrange the capabilities so the ones needed by the booter are past some "COMP_FREE" offset, then only llbooter will need to set it's captbl to the LLBOOTER_COMP_FREE and others can reuse those..

1. PTE caps are only required by the booter. 
2. COMP0 caps are not used at all in the new system in my understanding.. we could get rid of them.. 
3. Perhaps the fault capabilities are not required in the booter's captbl.. 
We can rearrange them I'm sure.. Can save some capabilities but honestly I'm thinking of it as more of a code cleanup..

Reply:
Not all `>32`. 
Ones with `PTE` in the enums are only populated by the kernel for the booter. I think they're not used by any component.. not even booter **uses** it.
`SELF_COMP`, I guess with capmgr as capability manager, I believe other components may not need it but I don't know if we want to get rid of it. 

I think booter is a special case and I think it is absolutely fine because I think we very well know that booter is a "special case". For ex, that `BOOT_CAPTBL_SINV_CAP` capability is not populated for booter but it just exists and wasting one slot. 

We're not going to have any **abstraction** that special cases booter really so I don't think having BOOTER_XX_CAP and COMP_XX_CAP is really harmful. I think it would self document better, in that it says, booter is indeed special and it has more "low level" capabilities. 

I don't have an argument for worthiness of the time and effort that would be put in but..